tools: add show region by keyspace-id or table-id

rleungx opened the pull request:
<!--

Thank you for working on PD! Please read PD's [CONTRIBUTING](https://github.com/tikv/pd/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.

PR Title Format:
1. pkg [, pkg2, pkg3]: what's changed
2. *: what's changed

-->

### What problem does this PR solve?

<!--

Please create an issue first to describe the problem.
There MUST be one line starting with "Issue Number:  " and 
linking the relevant issues via the "close" or "ref".
For more info, check https://github.com/tikv/pd/blob/master/CONTRIBUTING.md#linking-issues.

-->
Issue Number: ref #9707, cp https://github.com/tidbcloud/pd-cse/pull/355, close https://github.com/tikv/pd/issues/10531

### What is changed and how does it work?

<!--

You could use the "commit message" code block to add more description to the final commit message.
For more info, check https://github.com/tikv/pd/blob/master/CONTRIBUTING.md#format-of-the-commit-message.

-->

```commit-message
```

### Check List

<!-- Remove the items that are not applicable. -->

Tests

<!-- At least one of these tests must be included. -->

- Unit test

### Release note

<!--

A bugfix or a new feature needs a release note. If there is no need to give a release note, just leave it with the `None`.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

-->

```release-note
None.
```


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * CLI: `region keyspace id` accepts an optional table-id and a flexible limit, enforces stricter keyspace-ID validation with clearer error messages, and constructs either a table-specific key-range query or a limit-only query as appropriate.

* **Tests**
  * New tests capture outgoing requests to verify endpoint paths, query parameters (including computed key/end_key ranges) and argument validation/error messages.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rleungx commented on tools/pd-ctl/pdctl/command/region_command.go:
```diff
@@ -418,6 +419,71 @@ func showRegionsByKeysCommandFunc(cmd *cobra.Command, args []string) {
+		Use:   `keyspace-id <keyspace_id> [table-id <table_id>] [<limit>] [--jq="<query string>"]`,
+		Short: "show regions for keyspace or table",
+		Run:   showRegionsByKeyspaceTableIDCommandFunc,
+	}
```
good suggestion

rleungx committed:
pd-ctl: extend region keyspace lookup to table ranges

Reuse the existing region keyspace command instead of adding a parallel entry point for table-scoped lookups.

Add request-shape tests so the keyspace and table range queries stay aligned with the current CLI behavior.

Signed-off-by: Ryan Leung <rleungx@gmail.com>

rleungx committed:
pd-ctl: fix region keyspace table query escaping

Keep table range lookups aligned with url.Values encoding and reject keyspace IDs outside the supported 24-bit range.

Update the command tests to match the request shape and cover the out-of-range keyspace case.

Signed-off-by: Ryan Leung <rleungx@gmail.com>

rleungx committed:
pd-ctl: fix region keyspace static checks

Align the new region keyspace helpers with the repository static checks by fixing import ordering and using named return values.

This keeps the CI-only cleanup together with the helper logic and its direct test coverage.

Signed-off-by: Ryan Leung <rleungx@gmail.com>

bufferflies commented on tools/pd-ctl/pdctl/command/region_command_test.go:
```diff
@@ -0,0 +1,219 @@
+// Copyright 2025 TiKV Project Authors.
```
```suggestion
// Copyright 2026 TiKV Project Authors.
```

rleungx commented:
/retest

bufferflies commented on tools/pd-ctl/pdctl/command/region_command.go:
```diff
@@ -536,28 +537,62 @@ func NewRegionWithKeyspaceCommand() *cobra.Command {
 	r.AddCommand(&cobra.Command{
-		Use:   "id <keyspace_id> <limit>",
+		Use:   "id <keyspace_id> [table-id <table_id>] [<limit>]",
 		Short: "show region information for the given keyspace id",
```
should also modify the short description.

bufferflies reviewed (commented):
rest lgtm

rleungx committed:
pd-ctl: fix region keyspace command help text

Clarify that table-id is an optional filter on keyspace lookup and align the new test file header year with the current review update.

Ultraworked with [Sisyphus](https://github.com/code-yeongyu/oh-my-openagent)

Co-authored-by: Sisyphus <clio-agent@sisyphuslabs.ai>
Signed-off-by: Ryan Leung <rleungx@gmail.com>

rleungx commented:
/retest

rleungx commented:
/retest

rleungx commented:
/retest

bufferflies commented:
/retest

rleungx commented:
/retest

bufferflies commented:
/retest

bufferflies commented:
/retest

bufferflies commented:
/retest
